### PR TITLE
chore(react-dialog): migrate to new package structure

### PR DESCRIPTION
**Pull request by TristanWatanabe**
## Changes:
- Runs `migrate-converged-pkg` which reflects updates made in #24458.
	- Moves stories to `stories/` folder in the root of the package.
	- Collocates any e2e tests with the component implementation and jest tests.
	- removes `common/` folder and moves contents to new `testing/` folder 
	- adds `docs/` folder to package root.
	- moves `.mock.` files to new `testing/` subfolder.
	- moves any `assets` folder to appropriate folder.

## Issue:
- Part of #24129

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.935 kB`<br />`17.686 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.901 kB`<br />`52.991 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.446 kB`<br />`11.033 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `83.147 kB`<br />`24.799 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/176464a38a9b6210505d6950e8bd81df959427e0'>176464a38a9b6210505d6950e8bd81df959427e0</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/Avatar-mount_base.html">1275</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/Avatar-mount.html">1272</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/Button-mount_base.html">927</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/Button-mount.html">926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/FluentProvider-mount_base.html">1489</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/FluentProvider-mount.html">1490</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/FluentProviderWithTheme-mount_base.html">588</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/FluentProviderWithTheme-mount.html">586</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">535</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">547</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">578</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/MakeStyles-mount_base.html">1958</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/MakeStyles-mount.html">1966</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/SpinButton-mount_base.html">2339</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25523/perf-test-react-components/SpinButton-mount.html">2336</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-dialog/src/components/Dialog/Dialog.cy.tsx [hunk 1]
```diff
@@ -179,6 +179,7 @@ describe('Dialog', () => {
         }
       }, [open]);
       return (
+        //eslint-disable-next-line react/jsx-no-bind
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/components/Dialog/Dialog.cy.tsx:182]: @TristanWatanabe I don't expect this to be triggered for test files, can you please create an issue to disable this rule for tests?

File: change/@fluentui-react-dialog-66c4df04-b1d2-4abd-99fc-d6053ebdfe08.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "chore: Migrate to new package structure.",
+  "packageName": "@fluentui/react-dialog",
+  "email": "tristan.watanabe@gmail.com",
+  "dependentChangeType": "patch"
```
Comment (layershifter) [inline change/@fluentui-react-dialog-66c4df04-b1d2-4abd-99fc-d6053ebdfe08.json:6]:
```suggestion
  "type": "none",
  "comment": "chore: Migrate to new package structure.",
  "packageName": "@fluentui/react-dialog",
  "email": "tristan.watanabe@gmail.com",
  "dependentChangeType": "none"
```

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-dialog/src/components/Dialog/Dialog.cy.tsx (hunk 1 above)
Comment (TristanWatanabe) [reply packages/react-components/react-dialog/src/components/Dialog/Dialog.cy.tsx:182]: #25576

**Pull request merged**